cdap-security/src/test/java/co/cask/cdap/security/authorization/InMemoryAuthorizer.java

```diff
@@ -103,8 +101,10 @@ public void revoke(EntityId entity, Principal principal, Set<Action> actions) {
 
   @Override
   public void revoke(EntityId entity) {
-    for (Principal principal : table.row(entity).keySet()) {
-      getActions(entity, principal).clear();
+    if (privileges.containsKey(entity)) {
+      for (Principal principal : privileges.get(entity).keySet()) {
+        getActions(entity, principal).clear();
```

Comment:
Why not just remove the `entity` from the `privileges` map? Looping like this is not atomic, hence can have race.